dashboard/app/controllers/potential_teachers_controller.rb

```diff
@@ -28,6 +31,15 @@ def show
     params.permit([:name, :email, :script_id, :receives_marketing]).to_h
   end
 
+  def send_hoc_email(params)
+    name = current_user&.email || params[:email]
```

Comment:
This is what seemed to me to be a clean solution instead of having to obtain the user information on the front end. This function then doesn't care if we're dealing with a signed in or signed out user: it'll work either way